Commencement.Mvc/Controllers/ViewModels/RegistrationModel.cs

```diff
@@ -73,14 +73,20 @@ public static RegistrationModel Create(IRepository repository, IList<Ceremony> c
 
                 foreach (var major in student.Majors)
                 {
-                    Ceremony ceremony = GetCeremony(repository, major);
-
-                    if (ceremony != null && !participations.Any(a => a.Ceremony == ceremony))
+                    //Ceremony ceremony = GetCeremony(repository, major);
+                    var ceremonyList = ceremonies.Where(a => a.Majors.Contains(major)).ToList();
+                    if (ceremonyList != null)
                     {
-                        var part = CreateCeremonyParticipation(participations.Count, edit, student, major, ceremony, registration, null, repository, admin);
-                        if (part != null)
-                        {
-                            participations.Add(part);
+                        foreach (var ceremony in ceremonyList)
```

Comment:
In fact, just put the where clause in your foreach, removing the IF statement all together.